src/components/ColorPicker/components/ColorPickerItemComponent/ColorPickerItemComponent.jsx

```diff
@@ -55,35 +75,56 @@ const ColorPickerItemComponent = ({
+  }, [colorAsStyle, isColorPickerItem, shouldRenderIndicatorWithoutBackground]);
+
   return (
-    <Tooltip content={tooltipContent}>
-      <li
-        className={cx("monday-style-color-item-wrapper", {
-          "selected-color": isSelected,
-          active: isActive,
-          circle: colorShape === COLOR_SHAPES.CIRCLE
-        })}
-      >
-        <div className="feedback-indicator" />
-        <Clickable
-          ref={itemRef}
-          ariaLabel={color}
-          className={cx("color-item", `color-item-size-${colorSize}`, {
-            "color-item-text-mode": shouldRenderIndicatorWithoutBackground
+    <div className="monday-style-color-item-container">
+      <Tooltip content={tooltipContent}>
+        <li
+          className={cx("monday-style-color-item-wrapper", {
+            "selected-color": isSelected,
+            active: isActive,
+            circle: colorShape === COLOR_SHAPES.CIRCLE
           })}
-          style={{ background: shouldRenderIndicatorWithoutBackground ? "transparent" : colorAsStyle }}
-          onClick={onClick}
-          tabIndex="-1"
-          onMouseDown={onMouseDown} // this is for quill to not lose the selection
         >
-          <div className="color-indicator-wrapper" style={colorIndicatorWrapperStyle}>
-            {shouldRenderIcon && (
-              <Icon icon={shouldRenderSelectedIcon ? SelectedIndicatorIcon : ColorIndicatorIcon} ignoreFocusStyle />
-            )}
-          </div>
-        </Clickable>
-      </li>
-    </Tooltip>
+          <div className="feedback-indicator" />
+          <Clickable
+            ref={itemRef}
+            ariaLabel={color}
+            className={cx("color-item", `color-item-size-${colorSize}`, {
+              "color-item-text-mode": shouldRenderIndicatorWithoutBackground,
+              rainbow: isColorPickerItem
+            })}
+            style={{
+              background: backgroundStyle
+            }}
+            onClick={isColorPickerItem ? toggleColorPicker : onClick}
+            tabIndex="-1"
+            onMouseDown={onMouseDown} // this is for quill to not lose the selection
+          >
+            <div className="color-indicator-wrapper" style={colorIndicatorWrapperStyle}>
+              {shouldRenderIcon && (
+                <Icon icon={shouldRenderSelectedIcon ? SelectedIndicatorIcon : ColorIndicatorIcon} ignoreFocusStyle />
+              )}
+            </div>
+          </Clickable>
+        </li>
+      </Tooltip>
+      {isColorPickerItem && isSelected && (
+        <div className="clear-color-picker" onClick={() => clearSelectedColor(customColor)}>
+          <Icon icon={Delete} iconType={Icon.type.SVG} ignoreFocusStyle iconSize={14} />
+        </div>
+      )}
+      {isColorPickerItem && showColorPicker && (
+        <div className="custom-color-picker">
+          <ChromePicker color={{ hex: customColor }} onChange={newColor => chooseCustomColor(newColor.hex)} />
```

Comment:
consider using useMemo for the color, since otherwise a new object will be created on each render call, causing unnecessary renders of `ChromePicker`. (using `useCallback` for the onChange will also be nice 😄 )